File: app/src/main/java/com/example/ui/LoginActivity.java
package com.example.ui;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.WindowManager;
import android.widget.Toast;

import com.example.ui.RealmDB.User;
import com.google.android.material.textfield.TextInputEditText;

import java.util.Objects;

import io.realm.Realm;
import io.realm.RealmConfiguration;
import io.realm.RealmQuery;
import io.realm.RealmResults;


public class LoginActivity extends AppCompatActivity {


//    private Realm realm1;
//    private RealmConfiguration configuration;
    private SharedPreferences sharedPreferences;
    public final String sharedPrefs="Shared";
    private SharedPreferences.Editor editor;

    @SuppressLint("CommitPrefEdits")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_login);
//        realm1=Realm.getDefaultInstance();
        sharedPreferences=getSharedPreferences(sharedPrefs, Context.MODE_PRIVATE);
        editor=sharedPreferences.edit();
      if(sharedPreferences.contains("email"))
      {
          startActivity(new Intent(getApplicationContext(),Dashboard.class));
          finish();
      }
//        MyApplication.app.loginAsync(Credentials.anonymous(), new App.Callback<io.realm.mongodb.User>() {
//            @Override
//            public void onResult(App.Result<io.realm.mongodb.User> result) {
//                if(result.isSuccess()) {
//                    //configuration=new RealmConfiguration.Builder().deleteRealmIfMigrationNeeded().build();
//                    Log.d("SuxessL","Success");
//                    io.realm.mongodb.User user = MyApplication.app.currentUser();
//                    Long partitionValue = 123L;
//                    assert user != null;
//                    syncConfiguration = new SyncConfiguration.Builder(user, partitionValue).build();
//                    Realm.setDefaultConfiguration(syncConfiguration);
//                    Realm.getInstanceAsync(syncConfiguration, new Realm.Callback() {
//                        @Override
//                        public void onSuccess(Realm realm) {
//                            realm1=realm;
//                        }
//                    });
//                }
//                else
//                {
//                    Log.d("SuxessL","UNSuccess");
//                }
//
//            }
//        });

        findViewById(R.id.signup).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
               startActivity(new Intent(getApplicationContext(),Signup.class));
            }
        });
        findViewById(R.id.login).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                loginAuth();
                startActivity(new Intent(getApplicationContext(), Dashboard.class));
            }
        });
    }
//    public void loginAuth()
//    {
//        User user=null;
//        if(!String.valueOf(((TextInputEditText) findViewById(R.id.loginemail)).getText()).isEmpty()) {
//            user = realm1.where(User.class).equalTo("email", String.valueOf(((TextInputEditText) findViewById(R.id.loginemail)).getText())).findFirst();
//            //Toast.makeText(this, ""+user.getPassword(), Toast.LENGTH_SHORT).show();
//
////            assert user != null;
////            if (user!=null) {
//
//            Log.d("user",realm1.where(User.class).findAll().toString());
//            Log.d("emailc",String.valueOf(((TextInputEditText) findViewById(R.id.loginemail)).getText()));
//                if (user.getEmail().equals(Objects.requireNonNull(((TextInputEditText) findViewById(R.id.loginemail)).getText()).toString()) && (user.getPassword().equals(((TextInputEditText) findViewById(R.id.loginpass)).getText().toString())))
//                {
//                            startActivity(new Intent(getApplicationContext(), Dashboard.class));
//                            finish();
////                }
//            }
//        }
//        assert user != null;
////        startActivity(new Intent(getApplicationContext(), Dashboard.class));
////        finish();
//        Log.d("email",user.getEmail());
//        editor.putString("email",user.getEmail()).apply();
//
//
//    }
    @Override
    protected void onDestroy() {
        super.onDestroy();
//        realm1.close(); // Remember to close Realm when done.
    }

}

File: app/src/main/java/com/example/ui/LoginActivity.kt
package com.example.ui

import android.annotation.SuppressLint
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.WindowManager
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatButton
import com.example.ui.Dashboard
import com.example.ui.Model.ChatModel
import com.example.ui.Model.UserModel
import com.example.ui.databinding.ActivityLoginBinding
import com.google.android.gms.tasks.OnCompleteListener
import com.google.android.material.snackbar.Snackbar
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.ValueEventListener
import com.google.firebase.firestore.EventListener
import com.google.firebase.firestore.FirebaseFirestore
import java.util.concurrent.Executors


class LoginActivity : AppCompatActivity() {
    private lateinit var auth: FirebaseAuth
    //    private Realm realm1;
    //    private RealmConfiguration configuration;
    var sharedPreferences: SharedPreferences? = null
    val sharedPrefs = "Shared"
    lateinit var binding:ActivityLoginBinding
    private var editor: SharedPreferences.Editor? = null
    private lateinit var db:FirebaseFirestore
    @SuppressLint("CommitPrefEdits")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
        binding= ActivityLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)

        auth=FirebaseAuth.getInstance()
        db= FirebaseFirestore.getInstance()
        sharedPreferences = getSharedPreferences(sharedPrefs, MODE_PRIVATE)
        editor =  getSharedPreferences(sharedPrefs, MODE_PRIVATE).edit()
        if (sharedPreferences!!.contains("email")) {
            auth.signInWithEmailAndPassword(sharedPreferences!!.getString("email",null)!!,sharedPreferences!!.getString("password",null)!!).addOnCompleteListener(){
                if(it.isSuccessful){
                    val editor= getSharedPreferences("Shared", MODE_PRIVATE).edit()
                    editor.putString("UID",it.result?.user?.uid).apply()
//                    it.result?.user.
                    db.collection("users").whereEqualTo("UID",it.result?.user?.uid).get().addOnCompleteListener(OnCompleteListener {
                        it.result?.documents?.get(0)?.data?.get("FirstName")
                        editor.putString("FirstName", it.result?.documents?.get(0)?.data?.get("FirstName").toString()).apply()
                        editor.putString("LastName",it.result?.documents?.get(0)?.data?.get("LastName").toString()).apply()
                        startActivity(Intent(this,Dashboard::class.java))
                        finish()
                    }).addOnCanceledListener {
                        Log.d("ErrorMsg",it.exception?.message!!)
                    }
                }
                else{
                    Snackbar.make(findViewById(R.id.parent),"Login failed",Snackbar.LENGTH_LONG).show()
                }
            }
        }

        findViewById<TextView>(R.id.signup).setOnClickListener { startActivity(Intent(applicationContext, Signup::class.java)) }
        findViewById<AppCompatButton>(R.id.login).setOnClickListener { //                loginAuth();
            auth.signInWithEmailAndPassword(binding.loginemail.text.toString(),binding.loginpass.text.toString()).addOnCompleteListener(){
                if(it.isSuccessful){
                    val editor= getSharedPreferences("Shared", MODE_PRIVATE).edit()
                    editor.putString("UID",it.result?.user?.uid).apply()
                    db.collection("users").whereEqualTo("UID",it.result?.user?.uid).get().addOnCompleteListener(OnCompleteListener {
                        it.result?.documents?.get(0)?.data?.get("FirstName")
                        editor.putString("FirstName", it.result?.documents?.get(0)?.data?.get("FirstName").toString()).apply()
                        editor.putString("LastName",it.result?.documents?.get(0)?.data?.get("LastName").toString()).apply()
                        editor.putString("email",binding.loginemail.text.toString()).apply()
                        editor.putString("password",binding.loginpass.text.toString()).apply()
                        startActivity(Intent(this,Dashboard::class.java))
                        finish()
                    }).addOnCanceledListener {
                        Log.d("ErrorMsg",it.exception?.message!!)
                    }

                }
                else{
                    Snackbar.make(findViewById(R.id.parent),"Login failed",Snackbar.LENGTH_LONG).show()
                }
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        //        realm1.close(); // Remember to close Realm when done.
    }
}